mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAOV2.java

```diff
@@ -0,0 +1,360 @@
+    private final CassandraBlobsDAO blobsDAO;
+    private final PreparedStatement insert;
+    private final PreparedStatement delete;
+    private final PreparedStatement selectMetadata;
+    private final PreparedStatement selectHeaders;
+    private final PreparedStatement selectFields;
+    private final PreparedStatement selectBody;
+
+    @Inject
+    public CassandraMessageDAOV2(Session session, CassandraTypesProvider typesProvider, CassandraBlobsDAO blobsDAO) {
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.typesProvider = typesProvider;
+        this.blobsDAO = blobsDAO;
+        this.insert = prepareInsert(session);
+        this.delete = prepareDelete(session);
+        this.selectMetadata = prepareSelect(session, METADATA);
+        this.selectHeaders = prepareSelect(session, HEADERS);
+        this.selectFields = prepareSelect(session, FIELDS);
+        this.selectBody = prepareSelect(session, BODY);
+    }
+
+    private PreparedStatement prepareSelect(Session session, String[] fields) {
+        return session.prepare(select(fields)
+            .from(TABLE_NAME)
+            .where(eq(MESSAGE_ID, bindMarker(MESSAGE_ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(insertInto(TABLE_NAME)
+                .value(MESSAGE_ID, bindMarker(MESSAGE_ID))
+                .value(INTERNAL_DATE, bindMarker(INTERNAL_DATE))
+                .value(BODY_START_OCTET, bindMarker(BODY_START_OCTET))
+                .value(FULL_CONTENT_OCTETS, bindMarker(FULL_CONTENT_OCTETS))
+                .value(BODY_OCTECTS, bindMarker(BODY_OCTECTS))
+                .value(BODY_CONTENT, bindMarker(BODY_CONTENT))
+                .value(HEADER_CONTENT, bindMarker(HEADER_CONTENT))
+                .value(PROPERTIES, bindMarker(PROPERTIES))
+                .value(TEXTUAL_LINE_COUNT, bindMarker(TEXTUAL_LINE_COUNT))
+                .value(ATTACHMENTS, bindMarker(ATTACHMENTS)));
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(QueryBuilder.delete()
+                .from(TABLE_NAME)
+                .where(eq(MESSAGE_ID, bindMarker(MESSAGE_ID))));
+    }
+
+    public CompletableFuture<Void> save(MailboxMessage message) throws MailboxException {
+        try {
+            CompletableFuture<Optional<UUID>> bodyContent = blobsDAO.save(IOUtils.toByteArray(message.getBodyContent(), message.getBodyOctets()));
+            CompletableFuture<Optional<UUID>> headerContent = blobsDAO.save(IOUtils.toByteArray(message.getHeaderContent(), message.getFullContentOctets() - message.getBodyOctets()));
+
+            return bodyContent.thenCompose(bodyContentId -> headerContent.thenApply(headerContentId -> Pair.of(bodyContentId, headerContentId)))
+                    .thenCompose(pair -> cassandraAsyncExecutor.executeVoid(
+                            setTextualLineCount(
```

Comment:
There is a way to setToNull for TextualLineCount, I'm trying to find the way to merge into one preparedStatement